### PR TITLE
major variables add (GradientFeatures)

### LINKED ISSUE #89: Add full RCM mission to attributes
Currently, in L2 attrs there is : 
:missionName = "RADARSAT Constellation" ;

It must be
:missionName = "RADARSAT Constellation 1" ;
or 2 or 3

### PR DESCRIPTION
**Pull request by vincelhx**
- `gradientFeatures.py `
=> added heterogeneity mask (owiWindFilter) + 400VV & 800VH masks in xr_dataset
(Yuan Zhao 2021) 
=> added work on streaks from gradients (Koch 2004) with mulitple polarisation/windows_size/resolutions results
=> added a documentation to plot streaks 

- `inversion.py`
=> started a work to have a logger 
=> deleted some useless `.compute()` ?! maybe less time for computation 
=> added `owiAncillaryNrcs` and `owiAncillaryNrcs_cross` to simulate NRCS from ancillary wind using gmf 

- config
=> small changes in the config file : 
``` yaml
add_gradientsfeatures: True
add_nrcs_model: False ( do not use yet ) 
```
if not specified both are set to false 

- results saving 
owiWindFilter (mask) + 4 temporar variables are added in the main group 
a group is created for directions results : `owiWindStreaks` 

- I also had to change RCM output format 
- changed RCM sensor_longname to fix #89

### CODE REVIEW

File: grdwindinversion/utils.py [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+logger = logging.getLogger('grdwindinversion')
+# logger.addHandler(logging.NullHandler())
+logger.setLevel(logging.DEBUG)
+
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter(
+    '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+console_handler.setFormatter(formatter)
+logger.addHandler(console_handler)
```
Comment (agrouaze) [inline grdwindinversion/utils.py:16]: What is the rational of having 2 handlers rather than one?

**Review by agrouaze — APPROVED** [review]

**Comment by vincelhx** [comment]
we thought that inversion time was too long : sometimes 10 minutes for a EW 

I added a `xr_dataset.load()` at the begining of the use of xr_dataset. It looks to be 5x faster than without it. 
It forces the load of data in memory and executes differ operations before inversion and streaks operations

File: grdwindinversion/utils.py (hunk 1 above)
Comment (vincelhx) [reply grdwindinversion/utils.py]: changed

Comment (vincelhx) [reply grdwindinversion/utils.py]: also changed in xsarsea to 

``` python
logging.basicConfig(level=logging.INFO,
                    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
logger = logging.getLogger('xsarsea')
```

**Review by agrouaze — APPROVED** [review]

**Review by agrouaze — APPROVED** [review]

**Review by agrouaze — APPROVED** [review]

**Comment by Skealz** [comment]
The test fails because the meta.yaml forces to xarray 2024.2.0. See the PR I just merged a few hours ago. To make this PR pass, you must reproduce these changes.

File: grdwindinversion/gradientFeatures.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+            # Convert to meteo convention
+            streaks_smooth_mean = self.convert_to_meteo_convention(
+                streaks_smooth_mean)
+
+            # Set attributes
+            streaks_smooth_mean['angle'].attrs['description'] = 'Wind direction estimated from local gradient; histograms smoothed first, then mean computed'
+
+            return streaks_smooth_mean
+
+        except Exception as e:
```
Comment (agrouaze) [inline grdwindinversion/gradientFeatures.py:312]: This try except is huge, could it be reduced to critical part with the proper exception raised (eg ValueError, IOError,...)

File: grdwindinversion/gradientFeatures.py
Comment (agrouaze) [inline grdwindinversion/gradientFeatures.py:1]: many large try/except, is there a good reason for that?

**Review by agrouaze — CHANGES_REQUESTED** [review]
For such large addition (heterogeneity mask) it would be good to have some unit tests, or at least one if it gives a large coverage for added methods.

File: grdwindinversion/gradientFeatures.py (hunk 2 above)
Comment (vincelhx) [reply grdwindinversion/gradientFeatures.py:312]: #91

File: grdwindinversion/gradientFeatures.py
Comment (vincelhx) [reply grdwindinversion/gradientFeatures.py:1]: #91

**Pull request merged**